activerecord/lib/active_record/railtie.rb

```diff
@@ -370,7 +370,7 @@ class Railtie < Rails::Railtie # :nodoc:
           ActiveRecord.query_transformers << ActiveRecord::QueryLogs
           ActiveRecord::QueryLogs.taggings.merge!(
             application:  Rails.application.class.name.split("::").first,
-            pid:          -> { Process.pid },
+            pid:          -> { Process.pid.to_s },
```

Comment:
As mentioned above, since we're calling `val.gsub`, there is a requirement that each value must be a string. This change prevents an error when the `pid` tag is present.